internal/runners/branch/tree.go

```diff
@@ -17,108 +16,87 @@ type branchNode struct {
+	localBranchFormatting string = "[ACTIONABLE]%s[/RESET] [DISABLED](Current)[/RESET]"
+)
 
-func (bt *BranchTree) BuildFromBranches(branches mono_models.Branches) {
-	bt.rootBranches = getRootBranches(branches)
-	for _, branch := range bt.rootBranches {
-		bt.tree[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
+func NewBranchOutput(branches mono_models.Branches, localBranch string) *BranchOutput {
+	return &BranchOutput{
+		branches:    branches,
+		localBranch: localBranch,
 	}
 }
 
-func buildBranchTree(currentBranch *mono_models.Branch, branches mono_models.Branches) tree {
-	t := getChildren(currentBranch, branches)
-	for _, branch := range branches {
-		// Discard any branches without tracking information as we are only interested
-		// in child branches of the current branch
-		if branch.Tracks == nil {
-			continue
-		}
-
-		// Check that this branch is a child branch and recursively build its tree
-		if _, ok := t[branchNode{branch.Label, branch.BranchID.String()}]; ok {
-			t[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
-		}
+func (bt *BranchOutput) MarshalOutput(format output.Format) interface{} {
+	if format != output.PlainFormatName {
+		return branchListing(bt.branches, bt.localBranch)
 	}
-	return t
+	return branchTree(bt.branches, bt.localBranch)
 }
 
-func getRootBranches(branches mono_models.Branches) mono_models.Branches {
-	var rootBranches mono_models.Branches
-	for _, branch := range branches {
-		// Account for forked projects where the root branches contain
-		// a tracking ID that is not in the current project's branches
-		if branch.Tracks != nil && containsBranch(branch.Tracks, branches) {
-			continue
-		}
-		rootBranches = append(rootBranches, branch)
-	}
-	return rootBranches
-}
+func branchListing(branches mono_models.Branches, localBranch string) []string {
+	sort.Slice(branches, func(i, j int) bool {
+		return branches[i].Label < branches[j].Label
+	})
 
-func containsBranch(id *strfmt.UUID, branches mono_models.Branches) bool {
+	var branchNames []string
 	for _, branch := range branches {
-		if branch.BranchID.String() == id.String() {
-			return true
-		}
+		branchName := applyFormatting(branch.Label, localBranch)
+		branchNames = append(branchNames, branchName)
 	}
-	return false
+	return branchNames
 }
 
-func getChildren(branch *mono_models.Branch, branches mono_models.Branches) tree {
-	children := make(tree)
-	if branch == nil {
-		return children
+func branchTree(branches mono_models.Branches, localBranch string) string {
+	tree := make(tree)
+	for _, branch := range model.GetRootBranches(branches) {
+		tree[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
 	}
 
-	for _, b := range branches {
-		if b.Tracks == nil {
+	var levelsCompleted []int
+	rootBranches := model.GetRootBranches(branches)
+	return treeString(tree, rootBranches, localBranch, 0, levelsCompleted)
```

Comment:
This is doing too much. Be a little more functional about your logic. A function that marshals to a string should ONLY be doing that.